examples/proportion_test.py

```diff
@@ -16,76 +16,81 @@
 
 import argparse
 import os
-from typing import Any, Callable, Tuple
+from typing import Tuple
 
 import jax.numpy as np
 import numpyro
 import numpyro.distributions as dist
-import pandas as pd
 from jax import random
 from jax.scipy.special import expit
 from numpyro.diagnostics import hpdi
 from numpyro.infer import MCMC, NUTS
-from scipy import stats
 
 
-def make_dataset() -> Tuple[np.ndarray, np.ndarray]:
+def make_dataset(rng_key) -> Tuple[np.ndarray, np.ndarray]:
     """
-    Make simulated dataset where potential customers who get a 
+    Make simulated dataset where potential customers who get a
     sales calls have ~2% higher chance of making another purchase.
     """
-    made_purchase_got_called = stats.bernoulli.rvs(0.084, size=51342)
-    made_purchase_no_calls = stats.bernoulli.rvs(0.061, size=48658)
-    
+    key1, key2, key3 = random.split(rng_key, 3)
+
+    num_calls = 51342
+    num_no_calls = 48658
+
+    made_purchase_got_called = dist.Bernoulli(0.084).sample(key1, sample_shape=(num_calls,))
+    made_purchase_no_calls = dist.Bernoulli(0.061).sample(key2, sample_shape=(num_no_calls,))
+
     made_purchase = np.concatenate([made_purchase_got_called, made_purchase_no_calls])
-    
-    is_female = stats.bernoulli.rvs(0.5, size=100000)
-    got_called = np.concatenate([np.ones(51342), np.zeros(48658)])
-    design_matrix = np.hstack([np.ones((100000, 1)),
+
+    is_female = dist.Bernoulli(0.5).sample(key3, sample_shape=(num_calls + num_no_calls,))
+    got_called = np.concatenate([np.ones(num_calls), np.zeros(num_no_calls)])
+    design_matrix = np.hstack([np.ones((num_no_calls + num_calls, 1)),
                                got_called.reshape(-1, 1),
                                is_female.reshape(-1, 1)])
-    
+
     return design_matrix, made_purchase
 
 
 def model(design_matrix: np.ndarray, outcome: np.ndarray = None) -> None:
     """
     Model definition: Log odds of making a purchase is a linear combination
-    of covariates. Specify aNormal prior over regression coefficients.
+    of covariates. Specify a Normal prior over regression coefficients.
     :param design_matrix: Covariates. All categorical variables have been one-hot
         encoded.
     :param outcome: Binary response variable. In this case, whether or not the
         customer made a purchase.
     """
-    
-    # Use multivariate normal prior in case we want to add more covariates later.
+
     beta = numpyro.sample('coefficients', dist.MultivariateNormal(loc=0.,
                                                                   covariance_matrix=np.eye(design_matrix.shape[1])))
```

Comment:
nit: this can be faster if you replace `covariance_matrix` by `scale_tril`. It is still fine if you want to leave it as-is.